File: front/app/types/project.ts
export interface Project {
  id: number;
  name: string;
  description?: string;
  progress: number;
  status?: 'planning' | 'in-progress' | 'on-hold' | 'completed';
  priority?: 'low' | 'medium' | 'high';
  startDate?: string;
  dueDate?: string;
  createdAt?: string;
  color?: string;
}

File: MadeForYouMobile/app/src/main/java/com/nickolss/madeforyou/components/Project.kt
package com.nickolss.madeforyou.components

// Modelo para receber dados (Leitura)
data class Project(
    val id: Int? = null,
    val name: String,
    val description: String,
    val progress: Int, // 0 a 100
    val status: String, // "Planejamento", "Em Progresso", etc
    val priority: String,
    val startDate: String,
    val dueDate: String,
    val color: String? = null,// Cor padrão
    val userId: String? = null // Uso interno
)

// Modelo para ENVIAR dados (Escrita - JSON Limpo)
data class ProjectRequest(
    val name: String,
    val description: String,
    val progress: Int,
    val status: String,
    val priority: String,
    val startDate: String,
    val dueDate: String,
    val color: String
)